lrlex/src/lib/lexer.rs

```diff
@@ -423,7 +423,7 @@ if 'IF'
         assert_eq!(lexer.line_col(lexemes[2].start()), (2, 3));
         assert_eq!(lexer.surrounding_line_str(lexemes[0].start()), " a");
         assert_eq!(lexer.surrounding_line_str(lexemes[1].start()), "❤ b");
-        assert_eq!(lexer.surrounding_line_str(lexemes[1].start()), "❤ b");
+        assert_eq!(lexer.surrounding_line_str(lexemes[2].start()), "❤ b");
```

Comment:
Wait, should the last string only be `"b"`?

Reply:
No, because it's the whole line that that lexeme is contained in.

Reply:
Either I'm missing something, or the tests shouldn't have passed.

Reply:
Ah, wait, this is testing `surrounding_line_str`. Nevermind.

Reply:
Wouldn't it make more sense to test here if the lexemes are correct? e.g.
```
assert_eq!(lexemes[1], "❤");
assert_eq!(lexemes[2], "b");
```

Reply:
A unicode test for the pure lexer is a good idea -- I'll add that in.